src/v/kafka/server/replicated_partition.cc

```diff
@@ -234,6 +236,16 @@ ss::future<bool> replicated_partition::is_fetch_offset_valid(
         // retry linearizable barrier to make sure node is still a leader
         auto ec = co_await linearizable_barrier();
         if (ec) {
+            /**
+             * when partition is shutting down we may not be able to correctly
+             * validate consumer requested offset. We claim that the offset is
+             * valid not to force consumer to reset its offset.
+             */
+            if (
+              ec == raft::errc::shutting_down
+              || ec == cluster::errc::shutting_down) {
+                co_return true;
+            }
```

Comment:
The offset might be invalid, but we can't verify that. Since shutting down is normal, it's desired to not return something as heavy weight as `offset_out_of_range`--makes sense. But because things are shutting down, why should it be preferred to treat the potentially invalid offset as valid, rather than returning something like `NOT_LEADER_FOR_PARTITION` that the client would likely be getting if they continued forward?

Reply:
@mmaslankaprv

Reply:
this is done